### PR TITLE
Feature/tag selection composable

### PR DESCRIPTION
**Pull request by unglazedstamp**
The tag selection dialog composable, need the tag viewmodel merged first #111

### CODE REVIEW

**Review by ntissieres — CHANGES_REQUESTED** [review]
While the repository still not functional, I don't think it's a good idea to approve a code that would crash if used. It may be preferable to comment every use of the repository and replace it by an intermediate compilable version of it, in order to test it in the emulator to see if everything looks fine (more importantly for UI). For example, you could comment every call to the repository and put direct value instead, and accept a nullable repository for TagViewModel.

**Comment by unglazedstamp** [comment]
I agree with you. However this is addressed in #136 which replaces the remote database by a local storage, once it's done you will be able to launch the app and check the UI. If I hardcode tags now, I will have to revert these changes later, so I prefer to wait for the local storage to be merged so that we don't have to go back to each hardcoded composable of the project when we connect the remote database.

**Comment by ntissieres** [comment]
Good point. Considering the code, it seems perfect and ready to be merged after the #136

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (ntissieres) [inline app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt:42]: It would be better to have a fixed size Dialog (modifier.width(...).height(...)), because actually the size of the composable change each time to fit what to display

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (ntissieres) [inline app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt:44]: LazyColumn is more adapted than a Column for scrollable list, it is more optimized (https://stackoverflow.com/questions/73734918/android-compose-difference-between-lazycolumn-and-column-with-verticalscroll)

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (ntissieres) [inline app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt:54]: Maybe Modifier.height would make more sense here, padding add 5.dp on top and on bottom so 10.dp and is not really meant for that type of situation

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (ntissieres) [inline app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt:56]: I'm not certain it's really intuitive to click on the text to select a tag, I would say we're used to click on the row (text + arrow included) to dive into the subtag, and maybe have an "add" icon after the arrow to effectivelly select the tag

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (ntissieres) [inline app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt:89]: Maybe it's an error on the simple repository but the hasSubTags doesn't work, it still appears when we reach the "childest" tag (like Football) and reach a page with nothing

**Review by ntissieres — CHANGES_REQUESTED** [review]
Overall the tag selection works fine but some little changes still required

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt:44]: That's true, I fixed it

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt]: Removed it

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt]: Thanks for the suggestion, I will change that

File: app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/compose/components/TagSelectionDialog.kt:68]: I think it will  be confusing to add another Icon, but maybe I don't get it , we can discuss that after the next meeting

**Review by ntissieres — APPROVED** [review]
The changes are approved, nice work !

**Pull request merged**
